File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.java
import java.util.Arrays;

class Solution {
    public String solution(int[] numbers) {
        String[] arrayString = Arrays.stream(numbers)
                                      .mapToObj(String::valueOf)
                                      .toArray(String[]::new);
        
        Arrays.sort(arrayString, (a, b) -> (b + a).compareTo(a + b));
        
        //String result = String.join("", arrayString);
        
        StringBuilder resultArray = new StringBuilder();
        for (String result : arrayString ){
            resultArray.append(result);
        }
         String result = resultArray.toString();
        return result.charAt(0) == '0' ? "0" : result;
    }
}

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.py
def solution(numbers):
    # 양의 정수가 담긴 배열 numbers
    # 정수를 이어붙여 만들 수 있는 가장 큰수
    
    answer = ''
    numbers = list(map(str, numbers))
    numbers.sort(reverse=True, key=lambda x: x * 10)
    answer = ''.join(numbers)  
    
    return '0' if answer[0] == '0' else answer
